### ISSUE TITLE
(^)/2 operator is too defensive

### ISSUE THREAD

**Issue by jburse** [SWI-Prolog/swipl-devel#305]
SWI-Prolog can compute 9^(9^9) without a resource error and
very quickly ( :-) ), even if despite it has ca. 370 million digits:

```
   Welcome to SWI-Prolog (threaded, 64 bits, version 7.7.15)
   SWI-Prolog comes with ABSOLUTELY NO WARRANTY. This is free software.

   ?- [user].
   pow(_,0,1) :- !.
   pow(X,1,X) :- !.
   pow(X,N,Y) :-
        M is N//2, pow(X,M,A),
        J is N mod 2, pow(X,J,B),
        Y is (A*A)*B.
   ^D

   ?- pow(9,9,X).
   X = 387420489.

   ?- time((pow(9,9,X), pow(9,X,_))).
   % 159 inferences, 7.500 CPU in 8.514 seconds (88% CPU, 21 Lips)
```

**Comment by jburse** [comment]
But somehow the new ISO core corrigendum operator (^)/2
refuses to do it. Namely I get the following error:

```
    ?- _ is 9^(9^9).
    ERROR: Stack limit (1.0Gb) exceeded 
```

**Comment by JanWielemaker** [comment]
Thanks.  Fixed with 5af32264445d708aa055f231aadd7cd1ec42cfb5

**Comment by jburse** [comment]
Cool! I just noticed the ECLiPSe Prolog can also do it:
```

[eclipse 2]: _ is 9^(9^9).
Yes (20.09s cpu)
```

Very currious what the timing will be!
